.github/workflows/pipeline_set_output_tag.sh

```diff
@@ -21,5 +22,10 @@ if [ $github_ref_type == "tag" ]; then
   VERSION=$github_ref_name
 fi
 
+if [ $github_base_ref == "main" ]; then
```

Comment:
I don't see the source of the `github_base_ref` variable

Reply:
The runner sets it, and a bunch of other variables.

Reply:
But you define only the following variables: ```github_ref_type=$1
github_ref_name=$2
github_sha=$3```

github_base_ref undefined in here, no?

Reply:
Well the full case looks like this(also providing some context):
```
      - name: Setting a buildtag
        id: settag
        working-directory: .github/workflows
        run: ./pipeline_set_output_tag.sh $GITHUB_REF_TYPE ${{ github.ref_name }} ${{ github.sha }}
```

The variables here only expanded in various workflow action statements, for example: `run`. It is the first, second and third argument to that shellscript. Since workflow shell can be shaky at times when we do "magic" (additional processing to these variables) we opted for using a separate shellscript so workflow file can be more clean, and the processing part more separated/contained. the $1, $2, $3 are the arguments we pass along, but inside the script we name them again and finally "export" them after processing.

But now we are talking about it... Better add some safeties in the script.

Reply:
Added safety and yup you were right. I'll sit in the box of shame. ![image](https://user-images.githubusercontent.com/100154938/213116875-5e3feca8-6e39-4ca4-babe-15a5b257862f.png)